services/director-v2/src/simcore_service_director_v2/modules/docker_registry.py

```diff
@@ -0,0 +1,113 @@
+""" Module responsible of communicating with docker registry API
+
+"""
+# TODO: code below simply copied from old director and partially adapted
+# services/director/src/simcore_service_director/registry_proxy.py
+# services/director/src/simcore_service_director/registry_cache_task.py
+import logging
+from contextlib import suppress
+from typing import Dict, List
+
+from fastapi import FastAPI
+from httpx import AsyncClient
+
+from ..core.settings import RegistrySettings
+
+logger = logging.getLogger(__name__)
+
+# Module's setup logic ---------------------------------------------
+
+
+def setup(app: FastAPI, settings: RegistrySettings):
+    if not settings:
+        settings = RegistrySettings()
+
+    def on_startup() -> None:
+        app.state.docker_registry_api = RegistryApiClient(settings)
+
+    async def on_shutdown() -> None:
+        with suppress(AttributeError):
+            client: AsyncClient = app.state.docker_registry_api.client
+            await client.aclose()
+            del app.state.docker_registry_api
+
+    app.add_event_handler("startup", on_startup)
+    app.add_event_handler("shutdown", on_shutdown)
+
+
+# Module's business logic ---------------------------------------------
+#
+# TODO: this is totally unfinished!!
```

Comment:
Guess for now this will be included like this. If it must not be used, could you consider replacing the `pass` with a `NotImplementedError` or something similar?

Reply:
Done